compiler/aarch64/codegen/ARM64Instruction.hpp

```diff
@@ -188,29 +132,14 @@ class ARM64ImmInstruction : public TR::Instruction
    uint32_t setSourceImmediate(uint32_t si) {return (_sourceImmediate = si);}
 
    /**
-    * @brief Gets relocation kind
-    * @return relocation kind
-    */
-   TR_ExternalRelocationTargetKind getReloKind() { return _reloKind; }
-   /**
-    * @brief Sets relocation kind
-    * @param[in] reloKind : relocation kind
+    * @brief Encodes the immediate field into the instruction
+    * @param[in] instruction : instruction address
     */
-   void setReloKind(TR_ExternalRelocationTargetKind reloKind) { _reloKind = reloKind; }
-
-   /**
-    * @brief Gets symbol reference
-    * @return symbol reference
-    */
-   TR::SymbolReference *getSymbolReference() {return _symbolReference;}
-   /**
-    * @brief Sets symbol reference
-    * @param[in] sr : symbol reference
-    * @return symbol reference
-    */
-   TR::SymbolReference *setSymbolReference(TR::SymbolReference *sr)
+   virtual void insertImmediateField(uint32_t *instruction)
       {
-      return (_symbolReference = sr);
+      // Overlay the 32-bit immediate with no masking or shifting
+      //
+      *instruction |= _sourceImmediate;
```

Comment:
I think `*instruction = _sourceImmediate;` (no ORing) is better here.

Reply:
I had something in mind last Friday when I wrote this, but alas I can't think of a current, useful example where OR'ing would be the behaviour you want.  I'll change it to a store.  It's easy enough to change in the future should this not be the behaviour we want.

Reply:
Fixed in ff0a1face9a3264be38edd9e7b232d5457975c7b